client/src/main/java/io/pravega/client/stream/impl/LargeEventWriter.java

```diff
@@ -188,7 +188,7 @@ private void write(Segment parentSegment, List<ByteBuf> payloads, RawClient clie
         }
         requestId = client.getFlow().getNextSequenceNumber();
 
-        MergeSegments merge = new MergeSegments(requestId, created.getSegment(), parentSegment.getScopedName(), token);
+        MergeSegments merge = new MergeSegments(requestId, parentSegment.getScopedName(), created.getSegment(), token);
```

Comment:
@co-jo FYI even though this is the same change as the one in your integration test PR, you will get a merge conflict on this line.